src/System.IO/tests/BinaryWriter/BinaryWriter_WriteTests.cs

```diff
@@ -0,0 +1,744 @@
+
+            dw2.Flush();
+            mstr.Position = 0;
+            BinaryReader dr2 = new BinaryReader(mstr);
+
+            for (ii = 0; ii < sglArr.Length; ii++)
+            {
+                if ((sgl2 = dr2.ReadSingle()) != sglArr[ii])
+                    Assert.Fail("Error_48yf4_" + ii + "! Expected==" + sglArr[ii] + " , got==" + sgl2);
+            }
+
+            dw2.Dispose();
+            dr2.Dispose();
+            mstr.Dispose();
+        }
+
+        [TestMethod]
+        public static void BinaryWriter_WriteSingleTest_Negative()
+        {
+            Single sgl2 = 0;
+            int ii = 0;
+            Single[] sglArr = new Single[] { 
+                Single.MinValue, Single.MaxValue, Single.Epsilon, Single.PositiveInfinity, Single.NegativeInfinity, new Single(), 
+                0, (Single)(-1E20), (Single)(-3.5E-20), (Single)(1.4E-10), (Single)10000.2, (Single)2.3E30 
+            };
+
+            // [] read/Write with Memorystream
+            MemoryStream mstr = new MemoryStream();
+            BinaryWriter dw2 = new BinaryWriter(mstr);
+            for (ii = 0; ii < sglArr.Length; ii++)
+                dw2.Write(sglArr[ii]);
+
+            dw2.Flush();
+            mstr.Position = 0;
+            BinaryReader dr2 = new BinaryReader(mstr);
+            for (ii = 0; ii < sglArr.Length; ii++)
+                sgl2 = dr2.ReadSingle();
+
+            // [] Check End Of Stream
+            try
+            {
+                sgl2 = dr2.ReadSingle();
+                Assert.Fail("Error_2d847! Expected exception not thrown, sgl2==" + sgl2);
+            }
+            catch (IOException)
+            {
+            }
+            catch (Exception exc)
+            {
+                Assert.Fail("Error_238gy! Unexpected exception thrown, exc==" + exc.ToString());
+            }
+        }
+
+        [TestMethod]
+        public static void BinaryWriter_WriteDecimalTest()
+        {
+            Decimal dbl2 = 0;
+            int ii = 0;
+            Decimal[] dblArr = new Decimal[] { 
+                Decimal.One, Decimal.Zero, Decimal.MinusOne, Decimal.MinValue, Decimal.MaxValue, 
+                new Decimal(-1000.5), new Decimal(-10.0E-40), new Decimal(3.4E-40898), new Decimal(3.4E-28), 
+                new Decimal(3.4E+28), new Decimal(0.45), new Decimal(5.55), new Decimal(3.4899E23) 
+            };
+
+            // [] read/Write with Memorystream
+            MemoryStream mstr = new MemoryStream();
+            BinaryWriter dw2 = new BinaryWriter(mstr);
+
+            for (ii = 0; ii < dblArr.Length; ii++)
+                dw2.Write(dblArr[ii]);
+
+            dw2.Flush();
+            mstr.Position = 0;
+            BinaryReader dr2 = new BinaryReader(mstr);
+
+            for (ii = 0; ii < dblArr.Length; ii++)
+            {
+                if ((dbl2 = dr2.ReadDecimal()) != dblArr[ii])
+                    Assert.Fail("Error_948yg! Expected==" + dblArr[ii] + ", got==" + dbl2);
```

Comment:
Does Assert.Fail actually exist in xunit?  Regardless, this would be better as Assert.Equal, anyway, e.g.
```Assert.Equal(dblArr[ii], dr2.ReadDecimal());```